crates/jit/src/instantiate.rs

```diff
@@ -218,25 +226,33 @@ impl CompiledModule {
         Ok(Self {
             module: Arc::new(module),
             code: Arc::new(ModuleCode {
-                code_memory,
-                dbg_jit_registration,
+                code_memory: ManuallyDrop::new(code_memory),
+                dbg_jit_registration: ManuallyDrop::new(dbg_jit_registration),
+                image_range: image_range.0 as usize..image_range.0 as usize + image_range.1,
             }),
             finished_functions,
             trampolines,
             data_initializers,
             traps,
             stack_maps,
             address_transform,
-            obj,
             unwind_info,
         })
     }
 
     /// Extracts `CompilationArtifacts` from the compiled module.
     pub fn to_compilation_artifacts(&self) -> CompilationArtifacts {
+        // Get ELF image bytes and sanitize that.
+        let mut obj = Vec::from(unsafe {
```

Comment:
Would it be possible to not add `unsafe` here? Could the code bytes be read from a safe method?